### PR TITLE
HW_I2C Drivers

### PR DESCRIPTION
**Pull request by SoroushRohanizadeh**
### Summary
Created HW level I2C drivers and IO level drivers to speak with an I2C digital potentiometer.

### Changelist 
- added hw_i2c.c/h
- created vc/src/io folder
- added io_i2c_potentiometer.c/h in vc/src/io
- enabled I2C1 in the VC IOC

### Checklist
*Please change `[ ]` to `[x]` when you are ready.*
- [ x] I have read and followed the code conventions detailed in [README.md](../README.md) (*This will save time for both you and the reviewer!*).
- [ x] If this pull request is longer then **500** lines, I have provided *explicit* justification in the summary above explaining why I *cannot* break this up into multiple pull requests (*Small PR's are faster and less painful for everyone involved!*).

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -8,5 +8,13 @@
...
-	"cmake.configureOnOpen": false
+	"cmake.configureOnOpen": false,
+	"files.associations": {
+		"stdbool.h": "c",
+		"hw_i2c.h": "c",
+		"hw_hal.h": "c",
+		"io_led.h": "c",
+		"app_initstate.h": "c",
+		"io_i2c_potentiometer.h": "c"
+	}
```
Comment (Lucien950) [inline .vscode/settings.json:19]: NOOO!!!!!

Comment (Lucien950) [reply .vscode/settings.json:19]: please delete

File: firmware/shared/src/hw/hw_spi.h [hunk 2]
```diff
@@ -37,7 +37,7 @@ void hw_spi_setNssHigh(const SpiInterface *spi);
  * @return True if data is transmitted and received successfully. Else return
  * false.
  */
-bool hw_spi_transmitThenReceive(
+bool hw_spi_transmitAndReceive(
```
Comment (Lucien950) [inline firmware/shared/src/hw/hw_spi.h:40]: really hope this does not break random BMS code

File: firmware/shared/src/io/io_i2c_potentiometer.c [hunk 3]
```diff
@@ -0,0 +1,23 @@
+#include "io_i2c_potentiometer.h"
+#include "hw_i2c.h"
+
+const uint8_t TARGET_ADDRESS = 0b0101111;
+
+static I2cInterface *pot;
```
Comment (Lucien950) [inline firmware/shared/src/io/io_i2c_potentiometer.c:6]: bro literally did not hear my sumrath roast session

**Review by Lucien950 — CHANGES_REQUESTED** [review]

File: firmware/shared/src/hw/hw_spi.h (hunk 2 above)
Comment (SoroushRohanizadeh) [reply firmware/shared/src/hw/hw_spi.h:40]: Ok I don't even remember changing this so I'll change it back

File: firmware/shared/src/io/io_i2c_potentiometer.c (hunk 3 above)
Comment (SoroushRohanizadeh) [reply firmware/shared/src/io/io_i2c_potentiometer.c:6]: which line

File: .vscode/settings.json [hunk 4]
```diff
@@ -7,6 +7,5 @@
 			"url": "/can_bus/json/schema_tx.json"
 		}
 	],
-	"cmake.configureOnEdit": false,
-	"cmake.configureOnOpen": false
+	"cmake.configureOnEdit": false
```
Comment (Lucien950) [inline .vscode/settings.json:10]:
```suggestion
	"cmake.configureOnEdit": false,
	"cmake.configureOnOpen": false
```

File: firmware/quadruna/VC/src/io/io_i2c_potentiometer.c [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+const uint8_t TARGET_ADDRESS = 0b0101111;
+
+static I2cInterface pot;
+
+bool io_i2c_potentiometer_init(I2C_HandleTypeDef i2c_handle)
+{
+    pot.i2c_handle = &i2c_handle;
+    pot.target_address = TARGET_ADDRESS;
+    pot.timeout_ms = 100;
```
Comment (Lucien950) [inline firmware/quadruna/VC/src/io/io_i2c_potentiometer.c:12]:
```suggestion
extern I2C_HandleTypeDef _;

static I2cInterface pot {
    &_, TARGET_ADDRESS, 100
};

bool io_i2c_potentiometer_init()
{
```

**Review by Lucien950 — CHANGES_REQUESTED** [review]
one last thing

**Review by Lucien950 — APPROVED** [review]
image

**Pull request merged**